formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -245,6 +247,40 @@ internal class FormulaManagerImpl<Input, State, Output>(
         return manager.run(input).output
     }
 
+    fun <ChildInput, ChildOutput> child(
+        key: Any,
+        formula: IFormula<ChildInput, ChildOutput>,
+        input: ChildInput,
+        onError: (Throwable) -> Unit,
+    ): ChildOutput? {
+        val childrenManager = getOrInitChildrenManager()
+        val manager = childrenManager.findOrInitChild(key, formula, input)
+
+        // If termination happens while running, we might still be initializing child formulas. To
+        // ensure correct behavior, we mark each requested child manager as terminated to avoid
+        // starting new actions.
+        if (isTerminated()) {
+            manager.markAsTerminated()
+            manager.performTerminationSideEffects()
+        }
+        manager.setValidationRun(isValidationEnabled)
+
+        return try {
+            if (!manager.isTerminated()) {
+                manager.run(input).output
+            } else {
+                null
+            }
+        } catch (e: ValidationException) {
+            throw e
+        } catch (e: Throwable) {
+            manager.markAsTerminated()
+            onError(e)
+            manager.performTerminationSideEffects()
```

Comment:
performanceTerminationSideEffects could throw if the problematic code is within a transition. Not sure if that's something we should guard against.

Reply:
This is an excellent question that is a bit broader than this specific pull-request:
-  Currently, if `performTerminationSideEffects` crashes, we can leak memory by leaving an action(s) still running. 
- Related to this pull-request, if formula crashed, do we need to execute its remaining transitions? My take would be no